10_deploy_rook.sh

```diff
@@ -43,6 +43,10 @@ sed 's/namespace: rook-ceph/namespace: openshift-storage/' toolbox.yaml > toolbo
 sed -i "s@rook/ceph:master@rook/ceph:$ROOK_VERSION@" toolbox-modified.yaml
 oc create -f toolbox-modified.yaml
 
+oc wait --for condition=ready  pod -l app=rook-ceph-tools -n openshift-storage --timeout=180s
+oc -n openshift-storage exec $(oc -n openshift-storage get pod --show-all=false -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}') -- ceph mgr module enable pg_autoscaler
+oc -n openshift-storage exec $(oc -n openshift-storage get pod --show-all=false -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}') -- ceph config set global osd_pool_default_pg_autoscale_mode on
```

Comment:
need to restart the mon daemon here since we're using the 14.2.0 build.  i'm not sure what the best/right way to do that is tbh, @leseb?

Reply:
We should include that into Ceph config overrides, @umangachapagain please look into: https://rook.io/docs/rook/v0.9/advanced-configuration.html

Reply:
add something like this to the cluster.yaml:
```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: rook-config-override
  namespace: openshift-storage
data:
  config: |
    [global]
    osd_pool_default_pg_autoscale_mode = on
```

Reply:
@liewegas Will this also enable the module which was previously done by `ceph mgr module enable pg_autoscaler` ?
@leseb Should I restart the mon daemon? If yes, how?

Reply:
@umangachapagain you need **both** the `ceph mgr module enable pg_autoscaler` and the configmap.

You should inject the configmap before the injecting the cluster CR so you don't have to restart the mon.

Reply:
You mean,
Step 1. Inject configMap
Step 2. Deploy cluster
Step 3.  Deploy toolbox
Step 4. ` oc -n openshift-storage exec $(oc -n openshift-storage get pod --show-all=false -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}') -- ceph mgr module enable pg_autoscaler `

Reply:
Correct.

Reply:
Step 4 can be done at runtime right? I remember some commands can't be `exec`ed on runtime.

Reply:
Step 4 enables an mgr module, this can be done anytime the mgr is up and running, so no worries.

Reply:
Noted. Thanks.